treeherder/etl/job_loader.py

```diff
@@ -61,13 +63,16 @@ class JobLoader:
 
     def process_job(self, pulse_job, root_url):
         if self._is_valid_job(pulse_job):
+            acquire_connection()
```

Comment:
Yes. I can wrap the call to `Job_Loader().process_job` with the acquire and release in the ingestion script.

We need a semaphore here because we have `CONN_MAX_AGE` [here](https://github.com/mozilla/treeherder/blob/943a8ce1e591cd9e4cc6364bbcfcdb1a977d48cc/treeherder/config/settings.py#L129) set to a positive value.

So each time when a thread opens a connection to the DB, it is not closed until the connection exceeds 300 limits. This could be a problem if `process_job` is used somewhere where a persistent connection is needed.

Also I kind of arbitrarily picked 50 as the limit for `CONN_RESOURCES`. So right now there can only be 50 threads creating new connections with the DB. By default, MySQL allows 151 maximum connections, if we have multiple instances of the treeherder all processing ingestions this way, this number will probably need to be lowered.